src/ui/components/inputs/Checkbox/Checkbox.tsx

```diff
@@ -1,84 +1,243 @@
+  display,
+  hasError,
+  indeterminate,
   isChecked,
-  required,
+  label,
   onPress,
-  LabelComponent = StyledBody,
-  style,
+  required,
+  variant = 'default',
 }) => {
-  const { onFocus, onBlur, isFocus } = useHandleFocus()
+  const focusProps = useHandleFocus()
+  const hoverProps = useHandleHover()
+
+  const { designSystem, checkbox } = useTheme()
 
   const onToggle = useCallback(() => {
     onPress(!isChecked)
   }, [isChecked, onPress])
 
-  useSpaceBarAction(isFocus ? onToggle : undefined)
+  useSpaceBarAction(focusProps.isFocus ? onToggle : undefined)
+
+  const effectiveDisplay: CheckboxDisplay = display ?? (variant === 'detailed' ? 'fill' : 'hug')
+
+  const state = getCheckboxState(isChecked, indeterminate, hasError, disabled)
+  const isDisabled = state.includes('disabled')
+  const isCheckedState = state.includes('checked')
+  const isIndeterminate = state.includes('indeterminate')
+
+  const colorMark = disabled ? designSystem.color.icon.disabled : designSystem.color.icon.inverted
+  const CheckboxMarkIndeterminateSize = checkbox.size / 1.75 // Parent padding doesn't have effect on CheckboxMarkIndeterminate
 
   return (
     <CheckboxContainer
       {...accessibleCheckboxProps({ checked: isChecked, label, required })}
+      state={state}
+      variant={variant}
+      display={effectiveDisplay}
+      collapsed={collapsed}
       onPress={onToggle}
-      onFocus={onFocus}
-      style={style}
-      onBlur={onBlur}>
-      <Box isChecked={isChecked}>{isChecked ? <CheckboxMark width="80%" /> : null}</Box>
-      {label ? (
-        <LabelComponent>
-          {label}
-          {required ? '*' : null}
-        </LabelComponent>
-      ) : null}
+      {...focusProps}
+      {...hoverProps}>
+      <ContentContainer>
+        <LeftBox state={state} variant={variant}>
+          {isIndeterminate ? (
+            <CheckboxMarkIndeterminate
+              color={colorMark}
+              width={CheckboxMarkIndeterminateSize}
+              height={CheckboxMarkIndeterminateSize}
+            />
+          ) : isCheckedState ? (
+            <CheckboxMarkChecked color={colorMark} />
+          ) : null}
+        </LeftBox>
+        <RightBox>
+          <StyledBody disabled={isDisabled}>
+            {label}
+            {required ? '*' : null}
+          </StyledBody>
+          {description ? (
+            <StyledBodyAccentXs disabled={isDisabled}>{description}</StyledBodyAccentXs>
+          ) : null}
+        </RightBox>
+        {asset ? (
+          <BottomBox>
+            <CheckboxAsset {...asset} disable={isDisabled} />
+          </BottomBox>
+        ) : null}
+      </ContentContainer>
+      {collapsed ? <CollapsedContainer>{collapsed}</CollapsedContainer> : null}
     </CheckboxContainer>
   )
 }
 
-const CheckboxContainer = styled(TouchableOpacity)({
-  display: 'flex',
-  flexDirection: 'row',
+type ContainerProps = {
+  state: CheckboxState[]
+  variant: CheckboxVariant
+  collapsed?: React.ReactNode
```

Comment:
pourquoi pas un boolean ?

```suggestion
  collapsed: boolean
```

Reply:
C'est un composant qu'on veut ici